module/Company/src/Company/Controller/AdminController.php

```diff
@@ -0,0 +1,311 @@
+<?php
+
+namespace Company\Controller;
+
+use Zend\Mvc\Controller\AbstractActionController;
+use Zend\View\Model\ViewModel;
+
+class AdminController extends AbstractActionController
+{
+    public function indexAction()
+    {
+        $companyService = $this->getCompanyService();
+
+        $vm = new ViewModel(array(
+            'companyList' => $companyService->getHiddenCompanyList(),
+        ));
+
+        return $vm;
+    }
+
+    public function addPacketAction()
+    {
+        $companyService = $this->getCompanyService();
+        $packetForm = $companyService->getPacketForm();
+        $request = $this->getRequest();
+        $companyName = $this->params('slugCompanyName');
+        if ($request->isPost()) {
+            $companyService = $this->getCompanyService();
+            $packetForm->setData($request->getPost());
+
+            if ($packetForm->isValid()) {
+                $packet = $companyService->insertPacketForCompanySlugName($companyName);
+                $packet->exchangeArray($request->getPost());
+                $companyService->savePacket();
+
+                return $this->redirect()->toRoute('admin_company/editCompany', array('slugCompanyName' => $companyName), array(), false);
+            }
+        }
+        $packetForm->setAttribute('action',
+                                   $this->url()->fromRoute('admin_company/editCompany/addPacket',
+                                                           array('slugCompanyName' => $companyName))
+                                  );
+        $vm = new ViewModel(array(
+          //  'company' => $company,
+            'companyEditForm' => $packetForm,
+        ));
+
+        return $vm;
+    }
+    public function addCompanyAction()
+    {
+        $companyService = $this->getCompanyService();
+        $companyForm = $companyService->getCompanyForm();
+        $request = $this->getRequest();
+        if ($request->isPost()) {
+            if (!isset($companyName)) {
+                $companyName = $request->getPost()['slugName'];
+            }
+            $companyService = $this->getCompanyService();
+            $companyForm = $companyService->getCompanyForm();
+            $companyForm->setData($request->getPost());
+
+            if ($companyForm->isValid()) {
+                $company = $companyService->insertCompany($request->getPost()['languages']);
+                $company->exchangeArray($request->getPost());
+                $companyService->saveCompany();
+
+                return $this->redirect()->toRoute('admin_company/default', array('action' => 'edit', 'slugCompanyName' => $companyName), array(), false);
+            }
+        }
+        $companyForm->setAttribute('action',
```

Comment:
See http://www.php-fig.org/psr/psr-2/#4-6-method-and-function-calls how to format this part of the code